internal/storage/provider.go

```diff
@@ -22,13 +22,15 @@ type Provider interface {
 	FindIdentityVerification(ctx context.Context, jti string) (found bool, err error)
 
 	SaveTOTPConfiguration(ctx context.Context, config models.TOTPConfiguration) (err error)
+	UpdateTOTPConfigurationSignIn(ctx context.Context, config models.TOTPConfiguration) (err error)
 	DeleteTOTPConfiguration(ctx context.Context, username string) (err error)
 	LoadTOTPConfiguration(ctx context.Context, username string) (config *models.TOTPConfiguration, err error)
 	LoadTOTPConfigurations(ctx context.Context, limit, page int) (configs []models.TOTPConfiguration, err error)
 
-	SaveU2FDevice(ctx context.Context, device models.U2FDevice) (err error)
-	LoadU2FDevice(ctx context.Context, username string) (device *models.U2FDevice, err error)
-	LoadU2FDevices(ctx context.Context, limit, page int) (devices []models.U2FDevice, err error)
+	SaveWebauthnDevice(ctx context.Context, device models.WebauthnDevice) (err error)
+	UpdateWebauthnDeviceSignIn(ctx context.Context, device models.WebauthnDevice) (err error)
+	LoadWebauthnDevices(ctx context.Context, limit, page int) (devices []models.WebauthnDevice, err error)
```

Comment:
I think maybe the best way to handle this is to make a helper function, not entirely sure how you'd efficiently create an iterator within this context without loading every result to memory (the intention of this method is to avoid this and avoid potential issues).

Reply:
For reference, the only thing currently using it is the re-encryption method which loads 10 at a time to make the process relatively efficient.